src/main/java/com/marklogic/kafka/connect/sink/DefaultSinkRecordConverter.java

```diff
@@ -31,16 +35,30 @@ public DefaultSinkRecordConverter(Map<String, String> kafkaConfig) {
 		if (val != null && val.trim().length() > 0) {
 			format = Format.valueOf(val.toUpperCase());
 		}
-
 		val = kafkaConfig.get(MarkLogicSinkConfig.DOCUMENT_MIMETYPE);
 		if (val != null && val.trim().length() > 0) {
 			mimeType = val;
 		}
 	}
-
+	
+	/*
+	* method to check if the uses want the topic names to set as MarkLogic collections 
+	* @param SinkRecord 
+	* @return collections comma seperated collections string
+	*/
+	public String setCollection (SinkRecord sinkRecord){
+		if (addTopicIndicated) {
+			return  sinkRecord.topic() + ","+ mlcollection;
+	}
+	else {
+			return mlcollection;
+	}
+}
 	@Override
 	public DocumentWriteOperation convert(SinkRecord sinkRecord) {
-		return documentWriteOperationBuilder.build(toContent(sinkRecord));
+		return documentWriteOperationBuilder
```

Comment:
I believe this is where you need to add the topic to the list of collections. Instead of modifying the state of DocumentWriteOperationBuilder, which introduces a thread safety issue, I recommend doing the following:

- Call build
- Get the DocumentMetadataWriteHandle from the DocumentWriteOperation
- If it can be cast to a DocumentMetadataHandle, then do so and add the topic to the list of collections
- If it cannot be, log a warning and do nothing